### PR TITLE
Incorporate new div/sqrt unit

### PR DESCRIPTION
**Pull request by aswaterman**

### CODE REVIEW

File: src/main/scala/tile/FPU.scala [hunk 1]
```diff
@@ -241,9 +241,10 @@ trait HasFPUParameters {
   val (dExpWidth, dSigWidth) = (FType.D.exp, FType.D.sig)
   val floatTypes = FType.all.filter(_.ieeeWidth <= fLen)
   val maxType = floatTypes.last
-  def prevType(t: FType) = floatTypes(floatTypes.indexOf(t) - 1)
+  def prevType(t: FType) = floatTypes(typeTag(t) - 1)
   val maxExpWidth = maxType.exp
   val maxSigWidth = maxType.sig
+  def typeTag(t: FType) = floatTypes.indexOf(t)
```
Comment (hcook) [inline src/main/scala/tile/FPU.scala]: `typeTag` makes me think of http://docs.scala-lang.org/overviews/reflection/typetags-manifests.html, but it looks like we are no longer using the scala construct anywhere in the code base, so...ymmv.

File: src/main/scala/tile/FPU.scala [hunk 2]
```diff
@@ -764,45 +763,38 @@ class FPU(cfg: FPUParams)(implicit p: Parameters) extends FPUModule()(p) {
...
 
-    when (divSqrt_outValid) {
-      divSqrt_wen := !divSqrt_killed
-      divSqrt_wdata_double := sanitizeNaN(divSqrt.io.out, FType.D)
-      divSqrt_in_flight := false
-      divSqrt_flags_double := divSqrt.io.exceptionFlags
+    makeDivSqrt(FType.S, wb_ctrl.singleOut)
+    fLen match {
+      case 32 =>
+      case 64 => makeDivSqrt(FType.D, !wb_ctrl.singleOut)
```
Comment (hcook) [inline src/main/scala/tile/FPU.scala]: is this going to have 128 eventually?

Comment (aswaterman) [inline src/main/scala/tile/FPU.scala]: Yeah, for the Q extension.  For now, a match error for anything other than 32/64 is the desired behavior

File: src/main/scala/tile/FPU.scala (hunk 1 above)
Comment (aswaterman) [inline src/main/scala/tile/FPU.scala]: I was thinking of tagged unions when I wrote that

**Pull request merged**